designs/virtualization/4.x/vnc-in-new-window/vnc-in-new-window.md

```diff
@@ -0,0 +1,32 @@
+---
+parent: Virtualization
+version: 4.x
+---
+# VNC in new window
+
+## In the Administrator console
+
+In order to add an option to open the VNC console in a new window, we’ll add a new button ‘Open Console in new window’ to the existing ‘VNC console’ dropdown with the ‘Disconnect before switching’ checkbox option and ‘Send key’ button.
+
+![oprn from the new button](img/Open-console-1c.png)
```

Comment:
There were some early ideas around some text and iconography (long ago!!) for expanding this to a full window and opening in a new window/tab that I wanted to share incase it helps:
https://docs.google.com/document/d/1Fh2eb9TD0aq9JulVbLlss2ucdqYzx_uqTaThse_fcgE/edit#heading=h.3m45k77pubys

One thing I think this older design does is save some vertical space.

Reply:
Oh wow, blast from the past. 😄

I was going to suggest the same, maybe placing this button to the left of the `Send key` button to save some vertical space (should `Send key` be a dropdown by the way?). An external link icon may also be good here since a new tab/window opens when clicked. That old design was inspired by the actions in the Logs and Terminal tabs you can see in a Pod Details page today.

Reply:
Thanks @lizsurette and @andybraren.
The original design (as mentioned at the 'Note' in the google doc https://docs.google.com/document/d/1YFJDCWJj8NdodFhtXqq7KbhXz9EhHeBg9xyfybA0mVs/edit#heading=h.4kmd69x7yzsn) had all the buttons in one row to save some vertical space, but due to limitations of the buttons being part of the console component, we had to place the ‘Open Console in new window’ button in a separate row.
@glekner please correct me if anything changed around that.

Adding the external link icon is a good idea, thanks @andybraren.

Reply:
I don't think that part of the old design doc was ever implemented, but agreed, I'd be interested to hear if the component itself could be updated with the new button. Is the VNC Console component used anywhere else in the OpenShift UI? Maybe this is the only place? 🤷‍♂️

Reply:
> I don't think that part of the old design doc was ever implemented, but agreed, I'd be interested to hear if the component itself could be updated with the new button. Is the VNC Console component used anywhere else in the OpenShift UI? Maybe this is the only place? 🤷‍♂️

@yaacov any idea?

Reply:
Thanks for that old link @lizsurette I totally forgot about that :) 
I did confirm with the OCP stakeholders today that we don't have this surfacing anywhere else so it would be great to move this over to a PF4 approach. @glekner WDYT?

Reply:
Its doable, but it should be a different PR.

Reply:
+1 from me to move to PF4

Reply:
> Its doable, but it should be a different PR.

@glekner Yes, we discussed off-line that #5593 will use current component, and we will continue to make it closer to the design in follow-up PRs.